File: lib/java/projecteuler/Multiple.java
package projecteuler;

import static projecteuler.Series.sumNaturals;
import static projecteuler.Factor.isFactor;

public class Multiple {
  public static int multiplesUnder(int p, int n) {
    int pdn = p/n - (isFactor(n, p) ? 1 : 0);
    return n * sumNaturals(pdn);
  }
}

File: lib/scala/projecteuler/Multiple.scala
package projecteuler

import projecteuler.Series.sumNaturals

object Multiple {
  def multiplesUnder(p: Int, n: Int): Int = {
    val pdn = p/n - (if(p % n == 0) 1 else 0)
    return n * sumNaturals(pdn)
  }
}
